src/peft/tuners/lora/layer.py

```diff
@@ -260,6 +292,20 @@ def merge(self, safe_merge: bool = False, adapter_names: Optional[List[str]] = N
                     base_layer.weight.data = orig_weights
                 else:
                     base_layer.weight.data += self.get_delta_weight(active_adapter)
+
+                if (
+                    hasattr(base_layer, "_hf_hook")
+                    and isinstance(base_layer._hf_hook, AlignDevicesHook)
+                    and base_layer._hf_hook.offload
+                ):
+                    base_layer._hf_hook.weights_map = {
+                        name: param.to("cpu") for name, param in named_module_tensors(base_layer)
+                    }
+                    # offload weights map to disk if original device is the disk
+                    if torch.device("meta") in base_layer._hf_hook.original_devices.values():
+                        offload_folder = "offload_dir"
```

Comment:
Hmm, this creates a folder in the current working directory, right? This should be a temp folder instead.

Reply:
It probably should be, I have changed this to use a tempfile.TemporaryDirectory() instead.